static-site/components/list-resources/modal.tsx

```diff
@@ -0,0 +1,207 @@
+> | null>(null);
+
+/**
+ * A React Client Component that displays a provided `resource` in
+ * a modal popover window.
+ *
+ * @param resource - the VersionedResource object to display
+ */
+export function ResourceModal({
+  resource,
+}: {
+  resource: VersionedResource;
+}): React.ReactElement {
+  const [ref, setRef] = useState(null);
+
+  const setModalResource = useContext(SetModalResourceContext);
+  if (!setModalResource) {
+    throw new InternalError("Context not provided!");
+  }
+  const dismissModal: () => void = useCallback(() => {
+    setModalResource(undefined);
+  }, [setModalResource]);
+
+  // FIXME figure out what type can be assigned to `node`
+  // * `unknown` doesn't work
+  // * `React.SetStateAction<null>` doesn't work
+  // * `React.LegacyRef<HTMLDivElement>` doesn't work
+  const handleRef = useCallback((node): void => {
+    setRef(node);
+  }, []);
+
+  useEffect(() => {
+    function _handleEsc(event: KeyboardEvent): void {
+      if (event.key === "Escape") {
+        dismissModal();
+      }
+    }
+    window.addEventListener("keydown", _handleEsc);
+    return (): void => {
+      window.removeEventListener("keydown", _handleEsc);
+    };
+  }, [dismissModal]);
+
+  const scrollRef = useCallback((node: HTMLDivElement): void => {
+    // A CSS-only solution would be to set 'overflow: hidden' on
+    // <body>. This solution works well, but there are still ways to
+    // scroll (e.g. via down/up arrows)
+    node?.addEventListener(
+      "wheel",
+      (event: WheelEvent): void => {
+        event.preventDefault();
+      },
+      false,
+    );
+  }, []);
+
+  useEffect((): void => {
+    if (ref && resource) {
+      modal_draw(ref, resource, lightGrey);
+    }
+  }, [ref, resource]);
+
+  const summary = _snapshotSummary(resource.dates);
+
+  return (
+    <div ref={scrollRef}>
+      <div
+        className={styles.background}
+        style={{ backgroundColor: `${lightGrey}` }}
+        onClick={dismissModal}
+      />
+      <div
+        className={styles.modalContainer}
+        onClick={(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void => {
+          e.stopPropagation();
+        }}
+      >
+        <CloseIcon onClick={dismissModal} />
+        <div className={styles.title}>{resource.name.replace(/\//g, "│")}</div>
+        <div className={styles.snapshotWrapper}>
+          <strong>
+            {`${resource.dates.length} snapshots spanning ${summary.duration}: ${summary.first} - ${summary.last}`}
+          </strong>
+          <a
+            className={styles.snapshotLink}
+            href={`/${resource.name}`}
+            rel="noreferrer noopener"
+            target="_blank"
+          >
+            (click to view the latest available snapshot)
+          </a>
+        </div>
+        <div>{resource.updateCadence.description}</div>
+        {/* d3 controlled div */}
+        <div ref={handleRef} />
+        <div className={styles.italic}>
```

Comment:
_nitpicky nit_

The paragraphs in the modal don't have a space between sentences.

Reply:
fixed, thanks.